server/src/main/java/org/elasticsearch/index/mapper/DateFieldMapper.java

```diff
@@ -349,9 +350,17 @@ public Query rangeQuery(Object lowerTerm, Object upperTerm, boolean includeLower
                 throw new IllegalArgumentException("Field [" + name() + "] of type [" + typeName() +
                         "] does not support DISJOINT ranges");
             }
-            DateMathParser parser = forcedDateParser == null
-                    ? dateMathParser
-                    : forcedDateParser;
+            DateMathParser parser;
+            if (forcedDateParser == null) {
+                if (lowerTerm instanceof Integer || upperTerm instanceof Integer) {
```

Comment:
I think `Number` is better here. That, or `Long` and `Integer`. Its kind of plain luck that the range of numbers that cause the bug are small and so parsed as an `Integer` instead of a `Long` or whatever. But tests and stuff could still end up with a `Long`.